jax/_src/nn/initializers.py

```diff
@@ -122,11 +122,19 @@ def uniform(scale: RealNumeric = 1e-2,
   Array([[7.298188 , 8.691938 , 8.7230015],
          [2.0818567, 1.8662417, 5.5022564]], dtype=float32)
   """
+  use_default_dtype = dtype is None
+  if use_default_dtype:
+    dtype = jnp.float_
+
   def init(key: KeyArray,
            shape: core.Shape,
            dtype: DTypeLikeInexact = dtype) -> Array:
-    dtype = dtypes.canonicalize_dtype(dtype)
-    return random.uniform(key, shape, dtype) * lax.convert_element_type(scale, dtype)
+    if use_default_dtype:
```

Comment:
Should this be `if not use_default_dtype`?

Reply:
Made updates